src/screens/GroupInfo/hooks/useGroupInfo.js

```diff
@@ -265,43 +258,51 @@ const useGroupInfo = () => {
       user_id: item,
       channel_role: 'channel_moderator'
     }));
+    const user = {
+      id: profile?.myProfile?.user_id
+    };
+    const token = TokenStorage.get(ITokenEnum.token);
     await setOpenModal(false);
-    const filterMessage = await client.client.queryChannels(filter, sort, {
-      watch: true, // this is the default
-      state: true
-    });
-    navigation.reset({
-      index: 1,
-      routes: [
-        {
-          name: 'AuthenticatedStack',
-          params: {
-            screen: 'HomeTabs',
+    try {
+      await client.client.connectUser(user, token);
+      const filterMessage = await client.client.queryChannels(filter, sort, {
+        watch: true, // this is the default
+        state: true
+      });
+      navigation.reset({
+        index: 1,
+        routes: [
+          {
+            name: 'AuthenticatedStack',
             params: {
-              screen: 'ChannelList'
+              screen: 'HomeTabs',
+              params: {
+                screen: 'ChannelList'
+              }
+            }
+          },
+          {
+            name: 'AuthenticatedStack',
+            params: {
+              screen: 'ChatDetailPage'
             }
           }
-        },
-        {
-          name: 'AuthenticatedStack',
-          params: {
-            screen: 'ChatDetailPage'
-          }
-        }
-      ]
-    });
-    const generatedChannelId = generateRandomId();
-
-    if (filterMessage.length > 0) {
-      setChannel(filterMessage[0], dispatchChannel);
-    } else {
-      const channelChat = await client.client.channel('messaging', generatedChannelId, {
-        name: selectedUser.user.name,
-        type_channel: 1
+        ]
       });
-      await channelChat.create();
-      await channelChat.addMembers(memberWithRoles);
-      setChannel(channelChat, dispatchChannel);
+      const generatedChannelId = generateRandomId();
+      if (filterMessage.length > 0) {
+        setChannel(filterMessage[0], dispatchChannel);
+      } else {
+        const channelChat = await client.client.channel('messaging', generatedChannelId, {
+          name: selectedUser.user.name,
+          type_channel: 1
+        });
+        await channelChat.create();
+        await channelChat.addMembers(memberWithRoles);
+        setChannel(channelChat, dispatchChannel);
+      }
+    } catch (e) {
+      console.log({e, client}, 'error');
     }
   };
 
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

> Note: This review was outside of the patch, so it was mapped to the patch with the greatest overlap. Original lines [250-281]

The logic for opening a chat message has been updated. The user connection process has been added and the token is now being fetched from the `TokenStorage`. However, please ensure that the token is always available in the `TokenStorage` and handle cases where it might not be present.

```diff
- const filterMessage = await client.client.queryChannels(filter, sort, {
-   watch: true, // this is the default
-   state: true
- });
+ const user = {
+   id: profile?.myProfile?.user_id
+ };
+ const token = TokenStorage.get(ITokenEnum.token);
+ await setOpenModal(false);
+ try {
+   await client.client.connectUser(user, token);
+   const filterMessage = await client.client.queryChannels(filter, sort, {
+     watch: true, // this is the default
+     state: true
+   });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->